setup.cfg

```diff
@@ -67,13 +68,29 @@ dev =
 	%(test)s
 
 [flake8]
+exclude =
+    .git,
+    __pycache__,
+	env,
+    build,
+    dist,
+	nilearn/externals/tempita
+--select = D,E,F,W,C90
+docstring-convention = numpy
+max-line-length = 79
+max_complexity = 43
```

Comment:
I have set this to this value as it is currently the maximum of the most complex function / method.
This sets an upper bound to how complex the code can get before flake8 throws an error.
There is a handful of function that would need refactoring to bring this value back towards a more acceptable value (around 20 ish).